pkg/api/types.go

```diff
@@ -2226,7 +2226,10 @@ type ListOptions struct {
 	FieldSelector fields.Selector
 	// If true, watch for changes to this list
 	Watch bool
-	// The resource version to watch (no effect on list yet)
+	// For watch, it's the resource version to watch.
+	// For list, if unset (or zero), there is basically no guarantee on what List will return;
```

Comment:
Yes - the result will always contain resourceVersion of the returned result.

@hongchaodeng - I think I missed something when you asked me about it. So let me put this together again.
In cache the semantic is the following:
- if unset - redirect to etcd_helper, and then the result is returned from etcd based on quorum-read flag
- if 0 - then we simply return what we currently have, no guarantee
- in set to non zero - the result at least as fresh as given rv is returned

In all cases, the result contains RV from which it comes.

Can you please update this comment (sorry for confusion previously).